network.go

```diff
@@ -235,16 +209,50 @@ func (s *sandbox) updateInterface(netHandle *netlink.Handle, iface *pb.Interface
 	if lAttrs != nil && (lAttrs.Flags&net.FlagUp) == net.FlagUp {
 		// The link is up, makes sure we get it down before
 		// doing any modification.
-		if err := netHandle.LinkSetDown(link); err != nil {
-			return err
+		err = netHandle.LinkSetDown(link)
+		if err != nil {
+			goto error_case
 		}
 	}
 
-	if err := updateLink(netHandle, link, iface, actionType); err != nil {
-		return err
+	err = updateLink(netHandle, link, iface)
+	if err != nil {
+		goto error_case
+	}
+	err = netHandle.LinkSetUp(link)
+
+error_case:
+	// in the event that an error occurred during the interface update, make sure we return
+	// the resulting state instead of the requested state
+	if err != nil {
+		resultingIfc = getInterface(netHandle, link)
+	} else {
+		resultingIfc = iface
+	}
+
+	return resultingIfc, netHandle.LinkSetUp(link)
```

Comment:
Doesn't this mean we're calling `LinkSetUp()` twice for the non-error scenario?

Reply:
That is a bug!  Will fix